storage/storage.go

```diff
@@ -117,13 +116,22 @@ type Client struct {
 	// xmlHost is the default host used for XML requests.
 	xmlHost string
 	// May be nil.
-	creds *google.Credentials
+	creds *auth.Credentials
 	retry *retryConfig
 
 	// tc is the transport-agnostic client implemented with either gRPC or HTTP.
 	tc storageClient
 }
 
+// credsJSON returns the raw JSON of the Client's creds, or an empty slice
+// if no credentials JSON is available.
+func (c Client) credsJSON() []byte {
+	if c.creds != nil && len(c.creds.JSON()) > 0 {
+		return c.creds.JSON()
+	}
+	return []byte{}
```

Comment:
I think this would be nicer returning the creds and a bool so that callers can check `creds, ok := credsJSON()`
```suggestion
func (c Client) credsJSON() ([]byte, bool) {
	if c.creds != nil && len(c.creds.JSON()) > 0 {
		return c.creds.JSON(), true
	}
	return []byte{}, false
```

Reply:
SGTM!